lib/ico-validator/ico_validation.rb

```diff
@@ -0,0 +1,23 @@
+module IcoValidation
+  def self.valid_ico?(ico)
+    ico = ico.to_s
+    ico.length == 8 && ico.match(/^\d+$/) && last_number_valid?(ico)
+  end
+
+  private
+
+  def self.last_number_valid?(ico)
+    ico[7].to_i == calculate_valid_last_number(ico)
+  end
+
+  def self.calculate_valid_last_number(ico)
+    sum = (0..6).inject(0) { |sum, i| sum += ico[i].to_i * (8 - i) }
```

Comment:
Shadowing outer local variable - `sum`.<br>Useless assignment to variable - `sum`. Use just operator `+`.